File: problem045.scala
import BigInt._
object Main extends App {
val limit = 100000
val n = 40754
val t = (1 to limit) map (x=>BigInt(x)) map(x=>x*(x+1)/2) filter(_>n)
val p = (1 to limit) map (x=>BigInt(x)) map(x=>x*(3*x-1)/2) filter(_>n)
val h = (1 to limit) map (x=>BigInt(x)) map(x=>x*(2*x-1)) filter(_>n)
val r = p intersect h intersect t
val rr = r.sorted
println(rr)
}

File: python/problem045.py
#!/usr/bin/python

def intersect(a, b):
    return list(set(a) & set(b))

limit = 100000
t = map(lambda x: x*(x+1)/2, range(1,limit))
p = map(lambda x: x*(3*x-1)/2, range(1,limit))
h = map(lambda x: x*(2*x-1), range(1,limit))
r = intersect(t, p)
rr = intersect(r, h)
print (sorted(rr))
